cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -72,6 +72,12 @@
   // set of (EndPoint, Field)s which were generated by it.
   private Map<EndPointField, Set<EndPointField>> outgoingSummary;
 
+  // We maintain the set of fields dropped in this lineage; ones with incoming, but no outgoing.
+  private Set<String> droppedFields;
+
+  // for every field, which readOperation did it ultimately come from
+  private Map<String, ReadOperation> fieldSources;
```

Comment:
We shouldn't need this field to store the origin, all the trackback should happen using the same logic